### PR TITLE
Update cookbook

### PR DESCRIPTION
**Pull request by jonnybarnes**
Update the cookbook to be much more easily used across multiple Ttotara based projects.

Compatible with any recent version of Ubuntu (14.04, 16.04, 17.10 and 18.04).

Adds NodeJS, any version, but defaults to latest LTS release (as of 2018-06-29).

Allows parent project to install specific APT, PHP, or npm packages they may need.

### CODE REVIEW

File: recipes/nodejs.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#
+# Cookbook Name::cookbook_totara
+# Recipe:: nodejs
+#
+# Install Nodejs
+#
+
+node.default['nodejs']['install_method'] = 'binary'
```
Comment (michaelw90) [inline recipes/nodejs.rb]: We should only run this recipe if it's been enabled within the Chef Cookbook.

File: recipes/php.rb [hunk 2]
```diff
@@ -5,32 +5,45 @@
...
-  trusted      true
+    uri     'ppa:ondrej/php'
+    trusted true
 end
 
 include_recipe "php"
 include_recipe "php-fpm"
+
+Array(node["cookbook_totara"]["php_packages"]).each_with_index do |package_name, index|
+  package package_name do
```
Comment (michaelw90) [inline recipes/php.rb]: Is this not the same as the apt recipe that you've put in above?

File: templates/default/nginx_vhost_totara.conf.erb [hunk 3]
```diff
@@ -1,98 +0,0 @@
...
-
-server {
-
-    server_name <%= @hostname %>;
-    <% if @ssl_key && @ssl_cert -%>
-    listen <%= @listen_ssl_port %>;
-    <% else -%>
-    listen <%= @listen_port %>;
-    <% end -%>
-
```
Comment (michaelw90) [inline templates/default/nginx_vhost_totara.conf.erb]: Why are we deleting this file?

File: recipes/nodejs.rb (hunk 1 above)
Comment (michaelw90) [reply recipes/nodejs.rb]: Also, I think the Node recipes shoudl be ran separately in the run_list for the specific machine, not in this cookbook.

File: templates/default/nginx_vhost_totara.conf.erb (hunk 3 above)
Comment (jonnybarnes) [reply templates/default/nginx_vhost_totara.conf.erb]: its identical to `templates/nginx/nginx_vhost_totara.conf.erb` its seems redundant to have both

which would you keep?

File: recipes/php.rb (hunk 2 above)
Comment (jonnybarnes) [reply recipes/php.rb]: thinking about it yes, if there’s extra packages wanted from a repo, just use the apt recipe, will update now

File: recipes/nodejs.rb (hunk 1 above)
Comment (jonnybarnes) [reply recipes/nodejs.rb]: Done in latest commit, the `run_list` of the project’s `chef.json` needs to include `cookbook_totara::nodejs` if node is desired. I’ve tested this works.

File: templates/default/nginx_vhost_totara.conf.erb (hunk 3 above)
Comment (jonnybarnes) [reply templates/default/nginx_vhost_totara.conf.erb]: We ended up with a vanilla config file in two locations, and cabsci specific one.

You can have the config files anywhere, you just need to reference the correct location in the parent project’s `Vagrantfile` (or maybe `Berksfile`).

It made sense to me to delete the vanilla config on its own and leave a vanilla config and cabsci config both in `templates/nginx`

**Comment by jonnybarnes** [comment]
I left the node code in the cookbook, but that it would have to be called specifcally by the parent project. Is this no the change you wanted?

**Pull request merged**